source/common/config/grpc_mux_impl.cc

```diff
@@ -90,6 +90,15 @@ void GrpcMuxImpl::sendDiscoveryRequest(absl::string_view type_url) {
     }
   }
 
+  if (starting_) {
+    // On the initialization of the gRPC mux, load the persisted config, if available. If the xDS
```

Comment:
This seems like it could "waste" memory and cpu resources, especially if 1) the save data is stale, 2) the server is ready and can easily serve config.
Shouldn't this be part of a fall-back option?